### PR TITLE
Convert this repo into a vs-code plugin only repo and update.

### PR DESCRIPTION
**Pull request by ksdme**
Built on top of https://github.com/coala/coala-vs-code/pull/75.

Note: To prevent regressions and PR's with failing build this PR packs a few commits dropping the old files and adding updating the client. **Reviewers** note that the first two dropping commits are the cause of at least 45 file changes (deletions). So, this PR is still manageable and small.

### CODE REVIEW

**Comment by gaocegege** [comment]
@gitmate-bot ff

**Comment by jayvdb** [comment]
Needs rebase

**Comment by jayvdb** [comment]
can you do a new PR which only deletes the old langserver stuff

lets get that part out of the road, so we can focus on the other stuff

File: test/index.ts [hunk 1]
```diff
@@ -5,18 +5,19 @@
...
+// extension host can call to run the tests. The test runner is expected to use
+// console.log to report the results back to the caller. When the tests are
+// finished, return a possible error to the callback or null if none.
 
-var testRunner = require('vscode/lib/testrunner');
+let testRunner = require("vscode/lib/testrunner");
 
 // You can directly control Mocha options by uncommenting the following lines
-// See https://github.com/mochajs/mocha/wiki/Using-mocha-programmatically#set-options for more info
+// See https://bit.ly/2zmjU3c for more info
```
Comment (jayvdb) [inline test/index.ts]: no thanks :P .  increase the line lengths and/or add ignore regexes

File: src/extension.ts [hunk 2]
```diff
@@ -1,51 +1,114 @@
...
+} from "vscode-languageclient";
+
+const coalals = {
+    langs: [
+        "html",
+        "python",
+    ],
+
+    command: {
+        usage: resolve(__dirname, "../coalals/coala-ls.sh"),
```
Comment (jayvdb) [inline src/extension.ts]: Can we avoid a shell script?  Not always gonna work on Windows, the primary audience for this plugin

**Comment by jayvdb** [comment]
A rather ugly backtrace at https://travis-ci.org/coala/coala-vs-code/builds/415196366

File: tslint.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                },
+                "esSpecCompliant": true
+            }
+        ],
+        "no-var-requires": false,
+        "no-shadowed-variable": false,
+        "object-literal-shorthand": false
+    },
+    "rulesDirectory": []
+}
```
Comment (jayvdb) [inline tslint.json]: EOL at EOF

File: test/index.ts [hunk 4]
```diff
@@ -5,18 +5,19 @@
 // By default the test runner in use is Mocha based.
 //
 // You can provide your own test runner if you want to override it by exporting
-// a function run(testRoot: string, clb: (error:Error) => void) that the extension
-// host can call to run the tests. The test runner is expected to use console.log
-// to report the results back to the caller. When the tests are finished, return
-// a possible error to the callback or null if none.
+// a function run(testRoot: string, clb: (error:Error) => void) that the
```
Comment (jayvdb) [inline test/index.ts]: if we can avoid modifying this file, that would be better.

Add it to ignore list if you need to.

File: src/extension.ts [hunk 5]
```diff
@@ -1,51 +1,115 @@
...
-	};
-	const clientOptions: LanguageClientOptions = {
-		documentSelector: documentSelector,
-	}
-	return new LanguageClient(command, serverOptions, clientOptions).start();
+import {
+    Disposable,
+    ExtensionContext,
+    workspace,
+    WorkspaceConfiguration
```
Comment (jayvdb) [inline src/extension.ts]: trailing comma

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala-vs-code/pull/73/commits/b9739b30e74ba2705d86e9b75f4b20b8efe18a59 https://github.com/coala/coala-vs-code/pull/73/commits/9b6fd70e6de57884ee3b4250ae6c0d85aa147698 https://github.com/coala/coala-vs-code/pull/73/commits/76068466631a89e85444bf7f51d2fa1e4a4ae340 https://github.com/coala/coala-vs-code/pull/73/commits/c150d5be2ed9be1e42066b91bb0abba0614e705d https://github.com/coala/coala-vs-code/pull/73/commits/6aea5b870ecd01ae4f05b7189e19e5eca6ffe47a https://github.com/coala/coala-vs-code/pull/73/commits/05f6a221e57beeb04cdf185135d5fb6838357de1 https://github.com/coala/coala-vs-code/pull/73/commits/c330349e0d18d31e6ba9d86e9920d89d4ac3328d https://github.com/coala/coala-vs-code/pull/73/commits/e18fe3ffcdf187bba4f488fbaca3d4ade97b1e60 https://github.com/coala/coala-vs-code/pull/73/commits/1515455b76fc2146010a5e5018b59682e6a378eb

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**